beacon.go

```diff
@@ -17,110 +17,174 @@
 package beacon
 
 import (
+	"context"
 	"encoding/xml"
 	"errors"
-	"golang.org/x/oauth2/google"
-	"google.golang.org/api/genomics/v1"
-	"google.golang.org/appengine"
+	"fmt"
 	"net/http"
+	"os"
 	"strconv"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/appengine"
 )
 
 type beaconConfig struct {
-	variantSetIds []string
+	projectID string
+	tableId   string
 }
 
+const (
+	projectKey = "GOOGLE_CLOUD_PROJECT"
+	bqTableKey = "GOOGLE_BIGQUERY_TABLE"
+)
+
 var config = beaconConfig{
-	variantSetIds: []string{"3049512673186936334"},
+	projectID: os.Getenv(projectKey),
+	tableId:   os.Getenv(bqTableKey),
 }
 
 func init() {
 	http.HandleFunc("/", handler)
 }
 
-func requestToSearch(r *http.Request) (*genomics.SearchVariantsRequest, string, error) {
-	allele := r.FormValue("allele")
-	search := &genomics.SearchVariantsRequest{
-		ReferenceName: r.FormValue("chromosome"),
-		VariantSetIds: config.variantSetIds,
+func handler(w http.ResponseWriter, r *http.Request) {
+	if err := validateServerConfig(); err != nil {
+		writeError(w, err)
+		return
 	}
 
-	coord, err := strconv.ParseInt(r.FormValue("coordinate"), 10, 64)
+	refName, allele, coord, err := parseInput(r)
 	if err != nil {
-		return nil, "", err
+		writeError(w, err)
+		return
 	}
 
-	// Start is inclusive, End is exclusive.  Search exactly for coordinate.
-	search.Start = coord
-	search.End = search.Start + 1
+	ctx := appengine.NewContext(r)
+	exists, err := genomeExists(ctx, refName, allele, coord)
+	if err != nil {
+		writeError(w, err)
+		return
+	}
 
-	if search.ReferenceName == "" {
-		return nil, "", errors.New("missing reference name")
+	if err := writeResponse(w, exists); err != nil {
+		writeError(w, err)
+		return
 	}
-	if allele == "" {
-		return nil, "", errors.New("missing allele")
+}
+
+func genomeExists(ctx context.Context, refName string, allele string, coord int64) (bool, error) {
+	// Start is inclusive, End is exclusive.  Search exactly for coordinate.
+	query := fmt.Sprintf(`
+		SELECT count(v.reference_name) as count
+		FROM %s as v
+		WHERE reference_name='%s'
+			AND v.start <= %d AND %d < v.end
+	 	 	AND reference_bases='%s'
+		LIMIT 1`,
+		fmt.Sprintf("`%s`", config.tableId),
+		refName,
+		coord,
+		coord+1,
+		allele)
+
+	bqclient, err := bigquery.NewClient(ctx, config.projectID)
+	if err != nil {
+		return false, newDataAccessError("Creating a BigQuery client", err)
+	}
+	it, err := bqclient.Query(query).Read(ctx)
+	if err != nil {
+		return false, newDataAccessError("Querying database", err)
 	}
 
-	return search, allele, nil
+	type Result struct {
+		Count int
+	}
+	var result Result
```

Comment:
since you don't need the result type other places, you can just say

var result struct {
  Count int
}
...

Reply:
done